src/test/java/org/example/catcafereservation/ReservationMapperTest.java

```diff
@@ -0,0 +1,137 @@
+package org.example.catcafereservation;
+
+import com.github.database.rider.core.api.dataset.DataSet;
+import com.github.database.rider.core.api.dataset.ExpectedDataSet;
+import com.github.database.rider.spring.api.DBRider;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mybatis.spring.boot.test.autoconfigure.MybatisTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@DBRider
+@MybatisTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class ReservationMapperTest {
+
+    @Autowired
+    private ReservationMapper reservationMapper;
+
+    @Nested
+    class ReadTests {
+
+        @Test
+        @DataSet(value = "datasets/reservations.yml")
+        @Transactional
+        void 指定した予約番号の予約情報を取得できること() {
+            // Arrange
+            String reservationNumber = "01J2K2JKM8Y8QES70ZQ0S73JSR";
+
+            // Act
+            Optional<Reservation> actual = reservationMapper.findByReservationNumber(reservationNumber);
+
+            // Assert
+            assertThat(actual).hasValueSatisfying(reservation -> {
+                assertThat(reservation.getId()).isEqualTo(1);
+                assertThat(reservation.getName()).isEqualTo("名前はにゃんでも登録できちゃうにゃん太郎");
+                assertThat(reservation.getReservationDate()).isEqualTo(LocalDate.of(2024, 9, 20));
+                assertThat(reservation.getReservationTime()).isEqualTo(LocalTime.of(11, 30));
+                assertThat(reservation.getEmail()).isEqualTo("test@example.com");
+                assertThat(reservation.getPhone()).isEqualTo("02022222222");
+                assertThat(reservation.getReservationNumber()).isEqualTo(reservationNumber);
+            });
```

Comment:
imo
expected という期待値の変数を用意して検証してもいいかなと思います。
```suggestion
            assertThat(actual).isEqualTo(expected);
```

Reply:
ありがとうございます！
そっちの方が良いか～！と思ったのですが、そもそも`equals`と`hashCode`を実装していなかったのでもとに戻しました…🙇